apps/andi/test/integration/andi_web/live/access_group_live_view/edit_access_group_live_view_test.exs

```diff
@@ -352,6 +352,51 @@ defmodule AndiWeb.AccessGroupLiveView.EditAccessGroupLiveViewTest do
     end
   end
 
+  describe "emits a user_access_group_associate event" do
+    test "when the access group is saved, the association is updated", %{curator_conn: conn} do
+      user_one_subject_id = UUID.uuid4()
+
+      {:ok, user} =
+        Andi.Schemas.User.create_or_update(user_one_subject_id, %{
+          subject_id: user_one_subject_id,
+          email: "blahblahblah@blah.com",
+          name: "Someone"
+        })
+
+      user_id = user.id
+
+      access_group = create_access_group()
+      assert {:ok, view, html} = live(conn, "#{@url_path}/#{access_group.id}")
+
+      # add a new user to the access group
+      manage_users_button = find_manage_users_button(view)
+      render_click(manage_users_button)
+
+      # search for the user by name and select it
+      html = render_submit(view, "user-search", %{"search-value" => user.name})
+      select_user = element(view, ".modal-action-text", "Select")
+      html = render_click(select_user)
+      assert get_text(html, ".search-table") =~ user.name
+
+      # save the search
+      save_button = element(view, ".manage-users-modal .save-search", "Save")
+      html = render_click(save_button)
+
+      # verfy that the selected user appear in the users table
+      assert get_text(html, ".access-groups-sub-table") =~ user.name
+      assert get_text(html, ".access-groups-sub-table") =~ user.email
+
+      # save the changes to the access group
+      save_button = element(view, ".save-edit", "Save")
+      html = render_click(save_button)
+
+      eventually(fn ->
+        access_group = AccessGroups.get(access_group.id) |> Andi.Repo.preload(:users)
+        assert [%Andi.Schemas.User{id: ^user_id}] = access_group.users
```

Comment:
Could we assert that a log in the log events table was also added?

Do we have a testing pattern for that, not sure if it makes sense to add here but, would be good to have tests that ensure they're added somewhere.

Reply:
We can absolutely do this! I'll go ahead and spin up a card for this effort - I don't think it's something that's limited to this event - we probably want to add it for datasets, orgs, ingestions, etc. When I did the initial audit event, I added those tests to the API, but not to the UI.

Reply:
Created internal#667 to track this effort.